src/Symfony/Bridge/PhpUnit/ClockMock.php

```diff
@@ -73,10 +73,12 @@ public static function register($class)
     {
         $self = get_called_class();
 
-        $mockedNs = array(substr($class, 0, strrpos($class, '\\')));
-        if (strpos($class, '\\Tests\\')) {
-            $ns = str_replace('\\Tests\\', '\\', $class);
-            $mockedNs[] = substr($ns, 0, strrpos($ns, '\\'));
+        $testNs = substr($class, 0, strrpos($class, '\\'));
+        $mockedNs = array($testNs);
+        if (false !== ($pos = strpos($testNs, '\\Tests\\'))) {
+            $mockedNs[] = substr_replace($testNs, '\\', $pos, 7);
```

Comment:
this changes the semantic a bit (only the first `\Tests\` is replaced now). Let's be safe and keep the code asis for this first part...